core/src/martian/schema.cljc

```diff
@@ -71,9 +72,31 @@
                (make-schema ref-lookup param)}))
        (into {})))
 
-(defn- resolve-ref [ref-lookup ref]
-  (let [[_ category k] (re-find #"#/(definitions|parameters)/(.*)" ref)]
-    (get-in ref-lookup [(keyword category) (keyword k)])))
+(defn lookup-ref
+  [ref ref-lookup]
+  (if (string/starts-with? ref "#/")
+    (or (get-in ref-lookup (drop 1 (map keyword (string/split ref #"/"))))
+        (throw (ex-info "Cannot find reference"
+                        {:reference ref})))
+    (throw (ex-info "Non-local references are not supported yet. References should start ref with '#/'"
+                    {:reference ref}))))
+
+(defn resolve-ref-object
```

Comment:
this function follows through when you have `$ref` pointing to other `$ref`. This is possible following OpenAPI v3, but not Swagger. I believe this scenario was not covered previously, so I added it here.